src/components/VolunteerDashbord/IndividualVolunteers/index.js

```diff
@@ -0,0 +1,224 @@
+import React, { useEffect, useState } from "react";
+import axios from "axios";
+import { Link } from "react-router-dom";
+import { useSelector } from "react-redux";
+import { METHODS } from "../../../services/api";
+import { PATHS } from "../../../constant";
+import moment from "moment";
+import ReactPaginate from "react-paginate";
+import { useDebounce } from "use-debounce";
+import { BsArrowUpDown } from "react-icons/bs";
+
+import "./styles.scss";
+
+function VolunteerOverview(props) {
+  const [searchTerm, setSearchTerm] = useState("");
+  const [classes, setClasses] = useState([]);
+  const [slicedClasses, setSlicedClasses] = useState();
+  const user = useSelector(({ User }) => User);
+  const [pageNumber, setPageNumber] = useState(0);
+  const [sortMethod, setSortMethod] = useState("dsc");
+  const [debouncedText] = useDebounce(searchTerm);
+
+  const limit = 10;
+
+  const volunteerClassData = props.location.state.pass;
+  const pageCount = Math.ceil(volunteerClassData.classes.length / limit);
+  const changePage = ({ selected }) => {
+    setPageNumber(selected);
+  };
+
+  const firstIndex = (pageNumber + 1) * limit;
+  const lastIndex = volunteerClassData.classes.length;
+
+  const lang = { en: "English", hi: "Hindi" };
+  let language = "";
+  new Set(volunteerClassData.classes.map((item) => item.lang)).forEach(
+    (item) => (language = language + lang[item] + ", ")
+  );
+
+  const languageMap = {
+    hi: "Hindi",
+    te: "Telugu",
+    en: "English",
+    ta: "Tamil",
+  };
+
+  useEffect(() => {
+    const data = volunteerClassData.classes.filter((searchValue) => {
+      if (searchTerm == "") {
+        return searchValue;
+      } else if (
+        searchValue.title.toLowerCase().includes(searchTerm.toLowerCase())
+      ) {
+        return searchValue;
+      }
+    });
```

Comment:
What you have should work, but generally, the callback function to `filter` should always **explicitly** return a value and that value should always be a `Boolean` with `true` meaning "use" and `false` meaning "do **not** use" for the resulting filtered `Array`.

Currently, you're returning a truthy value (object `searchValue`) in the cases that there was no entered `searchTerm` or the `searchTerm` can be found in the title of `searchValue`.  You're not explicitly returning anything otherwise so in that case you're implicitly returning the falsy `undefined`.  Instead, I'd suggest having the callback just answer the question should I keep this? by answering yes, exactly when `searchTerm == "" || searchValue.title.toLowerCase().includes(searchTerm.toLowerCase()))`.  But actually the first part is unnecessary since every string contains the (lowercase version) of the empty string, so my suggestion would be to just use the following:

```suggestion
    const data = volunteerClassData.classes.filter((searchValue) =>
        searchValue.title.toLowerCase().includes(searchTerm.toLowerCase()));
```